### DISCUSSION TITLE
Install questions

### DISCUSSION THREAD

**Discussion by Nairou** [Divested-Mobile/DivestOS-Build#39] [General] [answered]
I'm trying to install DivestOS for the first time, on a Pixel 2 XL, and have some questions.

- The download page lists a "Download" link, as well as a "Fastboot" link. Both download zip files, but the contents are very different. I'm assuming the Download link gives the sideload zip, so what is the Fastboot zip for?
- The installation link for each device goes to the LineageOS site. The installation instructions there don't mention a fastboot zip, but do require installation of TWRP. Does DivestOS also require installation of TWRP?

**Comment by SkewedZeppelin** [comment, marked as answer]
You need 3 files.
TWRP is not supported.
The Lineage link is there due to legacy reasons.
There are steps here: https://divestos.org/index.php?page=bootloader

```
Enable 'Allow OEM unlocking' under 'Developer options' in Settings if available
Reboot to the bootloader via key combination or $ adb reboot bootloader
$ fastboot flashing unlock
$ fastboot update fastboot.zip
$ fastboot erase avb_custom_key
$ fastboot flash avb_custom_key avb_pkmd.bin
Reboot to recovery (use volume buttons to navigate if on or key combination if off)
$ adb sideload system.zip
factory reset while in recovery
reboot
```

**Comment by Nairou** [comment]
Thank you! That makes a lot more sense, I'm glad not to deal with TWRP.

All steps worked fine except the last. I reboot into recovery mode, select the option to install an update via ADB, and then run the `adb sideload` command. But nothing happens. The phone acts like it's ready to receive, but `adb sideload` never prints anything and never finishes.

I'm not sure what to try next. Could this be done with `adb push` and then have the phone install the update locally?

**Reply by Nairou** [reply]
Also, strangely: the first time I ran `fastboot update fastboot.zip` it automatically rebooted, and booted up into DivestOS, as if fully installed. I thought that was odd, since I hadn't sideloaded the other zip file yet...

**Reply by SkewedZeppelin** [reply]
Ah, I forgot to mention the --no-reboot fastboot option.
It is installed, but only the OTA ZIPs include firmware images, the fastboot images do not contain firmware.
Hence why a second flash is recommended.
If adb isn't detected perhaps your computer has an outdated adb or no udev rules/permission?
Alternatively you can just install the latest build via the Updater in the Settings app.

**Reply by Nairou** [reply]
Ah, that makes more sense now.

And using the updater in the settings was painless, so I think I'm good to go! Thanks again!
